frontend/src/container/LogsContextList/index.tsx

```diff
@@ -0,0 +1,207 @@
+import { Button, Typography } from 'antd';
+import RawLogView from 'components/Logs/RawLogView';
+import Spinner from 'components/Spinner';
+import { PANEL_TYPES } from 'constants/queryBuilder';
+import { FILTERS } from 'container/QueryBuilder/filters/OrderByFilter/config';
+import { useGetExplorerQueryRange } from 'hooks/queryBuilder/useGetExplorerQueryRange';
+import { useIsDarkMode } from 'hooks/useDarkMode';
+import { getPaginationQueryData } from 'lib/newQueryBuilder/getPaginationQueryData';
+import { memo, useCallback, useEffect, useMemo, useState } from 'react';
+import { Virtuoso } from 'react-virtuoso';
+import { ILog } from 'types/api/logs/log';
+import { Query, TagFilter } from 'types/api/queryBuilder/queryBuilderData';
+
+import {
+	getOrderByTimestamp,
+	INITIAL_PAGE_SIZE,
+	LOGS_MORE_PAGE_SIZE,
+} from './configs';
+import { EmptyText, ListContainer, ShowButtonWrapper } from './styles';
+
+interface LogsContextListProps {
+	isEdit: boolean;
+	query: Query;
+	log: ILog;
+	order: string;
+	filters: TagFilter | null;
+}
+
+function LogsContextList({
+	isEdit,
+	query,
+	log,
+	order,
+	filters,
+}: LogsContextListProps): JSX.Element {
+	const isDarkMode = useIsDarkMode();
+	const [logs, setLogs] = useState<ILog[]>([]);
+	const [page, setPage] = useState<number>(1);
+
+	const firstLog = useMemo(() => logs[0], [logs]);
+	const lastLog = useMemo(() => logs[logs.length - 1], [logs]);
+	const orderByTimestamp = useMemo(() => getOrderByTimestamp(order), [order]);
+
+	const currentStagedQueryData = useMemo(() => {
+		if (!query || query.builder.queryData.length !== 1) return null;
+
+		return query.builder.queryData[0];
+	}, [query]);
+
+	const getRequestData = useCallback(
```

Comment:
responsibility for this function is get updated query can we move this one to util